File: webflux-dialogue/src/main/java/com/study/webflux/rag/application/monitoring/monitor/DialoguePipelineMonitor.java
package com.study.webflux.rag.application.monitoring.monitor;

import java.time.Clock;

import org.springframework.stereotype.Component;

@Component
public class DialoguePipelineMonitor {

	private final PipelineMetricsReporter reporter;
	private final Clock clock;

	public DialoguePipelineMonitor(PipelineMetricsReporter reporter, Clock clock) {
		this.reporter = reporter;
		this.clock = clock;
	}

	/**
	 * 입력 텍스트 기준으로 파이프라인 추적기를 생성합니다.
	 */
	public DialoguePipelineTracker create(String inputText) {
		return new DialoguePipelineTracker(inputText, reporter, clock);
	}
}

File: backend/monitoring/src/main/kotlin/com/miyou/app/monitoring/monitor/DialoguePipelineMonitor.kt
package com.miyou.app.monitoring.monitor

import com.miyou.app.monitoring.port.PipelineMetricsReporter
import org.springframework.stereotype.Component
import java.time.Clock

@Component
class DialoguePipelineMonitor(
    private val reporter: PipelineMetricsReporter,
    private val clock: Clock,
) {
    fun create(
        sessionId: String?,
        userId: String?,
        personaId: String?,
        inputText: String?,
    ): DialoguePipelineTracker = DialoguePipelineTracker(sessionId, userId, personaId, inputText, reporter, clock)
}
